libs/common/src/main/java/org/opensearch/common/collect/List.java

```diff
@@ -34,6 +34,13 @@
 
 import java.util.Collection;
 
+/**
+ * Java 9 List
+ *
+ * todo: deprecate and remove w/ min jdk upgrade to 11?
+ *
+ * @opensearch.internal
```

Comment:
This is a nitpick, but I think this is a bit of an abuse of the `@opensearch.internal` annotation. This was clearly meant to be a public utility just like anything else in the common namespace. It just has been superseded by an alternative, which is exactly what the `java.lang.Deprecated` annotation is meant to convey.